.github/workflows/deploy-docker-compose.yml

```diff
@@ -0,0 +1,161 @@
+name: Deploy Docker Compose
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        type: string
+        description: "The deployment environment (e.g., production, staging)"
+        required: true
+      docker-compose-file:
+        type: string
+        default: "./docker-compose.yml"
+        description: "Path to the Docker Compose file (Default: ./docker-compose.yml)"
+      main-image-name:
+        type: string
+        description: "The name of the main image for checking if it exists with the given tag"
+        required: true
+      image-tag:
+        type: string
+        description: "Image tag to deploy (default: pr-<number> if PR exists, latest for default branch)"
+      env-vars:
+        type: string
+        description: "Additional environment variables in KEY=VALUE format, separated by newlines"
+        required: false
```

Comment:
In my opinion, getting `env-vars` like a plain input is dangerous and it makes this reusable component useless for some projects where we still keep our secrets in `.env` file (Example projects following the same logic [Helios](https://github.com/ls1intum/Helios/blob/main/.github/workflows/deploy_docker.yml#L56-L87), [prompt2](https://github.com/ls1intum/prompt2/blob/main/.github/workflows/deploy-docker.yml#L62-L92), [thesis-track](https://github.com/ls1intum/thesis-track/blob/main/.github/workflows/deploy_docker.yml#L83-L138)).

I forked this `.github` repository and tried to set this `env-vars` input as secret and also searched a little bit and found that the `with` clause don't have support for `secrets`.
And also github shows all the inputs as plain text in the github actions:
![image](https://github.com/user-attachments/assets/d5f067b3-c720-42a4-af85-abd5d3780732)

![image](https://github.com/user-attachments/assets/606008e5-d2b5-48bf-93c8-c355b5bbad3a)


So as a solution I think we should use `secrets`. 
Example repository:
[Caller action](https://github.com/ege-test-org/.github/blob/main/.github/workflows/test.yml#L6-L7), [reusable action](https://github.com/ege-test-org/.github/blob/main/.github/workflows/deploy-docker-compose.yml#L35-L36)

And with this the logs become hidden
![image](https://github.com/user-attachments/assets/e506750e-5584-4ecb-98f5-957d8643e399)